components/ee/agent-smith/pkg/agent/agent.go

```diff
@@ -313,37 +319,74 @@ func (agent *Smith) Start(callback func(InfringingWorkspace, []PenaltyKind)) {
 
 	defer abpf.Close()
 
+	agent.cleanupDeadPIDS(ctx)
+
+	egressTicker := time.NewTicker(30 * time.Second)
+
 	for i := 0; i < 10; i++ {
 		go func(i int) {
-			for h := range agent.perfHandler {
-				if h == nil {
-					continue
-				}
+			for {
+				select {
+				case <-egressTicker.C:
+					agent.pidsMap.Range(func(key, value interface{}) bool {
+						p := key.(int)
+						t := value.(time.Time)
+						infr, err := agent.checkEgressTrafficCallback(p, t)
+						if err != nil {
+							return true
```

Comment:
I reckon at least some debug logging would be handy here

Reply:
A log here would be extremely verbose, I can add one with a non logged by default priority.
Keep in mind that this is done for every found process every 30 seconds.

Reply:
Excellent point. We could add a metric here also, to see how these checks behave, akin to the signature check failures.